.github/workflows/data_update_v2.yml

```diff
@@ -59,22 +59,15 @@ jobs:
       run: |
         echo "::set-env name=SCRAPER_TIME::$(date)"
         cd ${GITHUB_WORKSPACE}/scraper
+
         OUT_PATH="${GITHUB_WORKSPACE}/site/data/data.v2.json"
-        
-        # Ideally, we'd just run the scraper against all counties like below:
-        #     python scraper_data.py > "${OUT_PATH}"
-        # But we don't want to stop if there are errors, so instead we run each
-        # scraper individually. Then we combine all the outputs and overlay
-        # them on top of the previoius output.
-        (python scraper_data.py alameda || echo '{}') > alameda.json
-        (python scraper_data.py san_francisco || echo '{}') > san_francisco.json
-        (python scraper_data.py solano || echo '{}') > solano.json
-        (python scraper_data.py sonoma || echo '{}') > sonoma.json
-        # Combine outputs with the previous run's saved output:
-        jq -s \
-          '.[0] + .[1] + .[2] + .[3] + .[4]' \
-          "${OUT_PATH}" alameda.json san_francisco.json solano.json sonoma.json \
-          > "${OUT_PATH}"
+
+        python scraper_data.py > today.json 2> errors.txt
```

Comment:
The default shell for `run` commands in GitHub actions uses `set -e o pipefail`, which causes the shell to exit as soon as a command exits with a non-zero code. Since the scraper is going to set a non-zero exit code, this now needs to handle that.

There are a few good ways:

1. Use `|| true` at the end of the line to ignore the failure. This is basically making an exception for just this one line, and I think it’s a little nicer:

    ```sh
    python scraper_data.py > today.json 2> errors.txt || true
    ```

2. Switch the whole `run` step to a different mode:

    ```
    - name: Install Data Scraper & Dependencies
      shell: bash {0}
      run: |
        # The same run step as you currently have here
    ```

The other thing you could do here is capture the exit code so we can cause the job to fail later, *after* the step where we create the PR (so if the script had some successes and some errors, we still make the PR for the successes). For example, with version 1 above:

```
        # Run in a subshell and output the exit code (instead of exiting
        # with it, so it doesn't qualify as an error).
        CODE="$(
            python scraper_data.py > today.json 2> errors.txt
            printf "${?}"
        )"
        # Save the exit code for later use in another job step.
        echo "::set-env name=SCRAPER_EXIT_CODE::$(CODE)"

    # ...then in a last step, AFTER the "Create PR if New Data" step
    # so that we don't stop the whole job before creating the PR
    - name: Set Overall Exit Code for Action
      run: |
        if [ "${SCRAPER_EXIT_CODE}" -gt 0 ]; then false; fi
    ```

Reply:
Great catch! I went with option 1 as it is the simplest. I'd like to do handling total failures as suggested as a followup once we know this is working.